corehq/apps/api/resources/auth.py

```diff
@@ -40,7 +46,60 @@ def _inner(req, domain, *args, **kwargs):
     return _inner
 
 
-class LoginAndDomainAuthentication(Authentication):
+class HQAuthenticationMixin:
+    decorator_map = {}  # should be set by subclasses
+
+    def _get_auth_decorator(self, request):
+        return self.decorator_map[determine_authtype_from_header(request)]
+
+    def get_identifier(self, request):
+        username = request.couch_user.username
+        if API_THROTTLE_WHITELIST.enabled(username):
+            return username
+        return f"{getattr(request, 'domain', '')}_{username}"
+
+
+class LoginAuthentication(HQAuthenticationMixin, Authentication):
+    """
+    Just checks you are able to login. Does not check against any permissions/domains, etc.
+    """
+    def __init__(self):
+        super().__init__()
+        self.decorator_map = {
+            'digest': digest_auth_no_domain,
+            'basic': basic_auth_no_domain,
+            'api_key': api_key_auth_no_domain,
+            'oauth2': oauth2_auth_no_domain,
+        }
+
+    def is_authenticated(self, request, **kwargs):
+        return self._auth_test(request, wrappers=[
+            self._get_auth_decorator(request),
+            wrap_4xx_errors_for_apis,
+        ], **kwargs)
+
+    def _auth_test(self, request, wrappers, **kwargs):
+        PASSED_AUTH = 'is_authenticated'
```

Comment:
Nit: could make it nearly impossible for an wrapper to return a value that is interpreted as "is_authenticated". Currently it is easy, if possibly not likely.
```suggestion
        PASSED_AUTH = object()
```